### PR TITLE
Refined claim criteria

### PR DESCRIPTION
**Pull request by schmidtd**
Implementing https://github.com/open-cluster-management/backlog/issues/5590, noting that the second part of the criteria - i.e. not in a hibernated state - is actually covered by the initial criteria.

### CODE REVIEW

**Comment by gurnben** [comment]
### TLDR
The cluster status check (`The cluster is then running when, in the ClusterDeployment, the .status.conditions | select(.type == "Hibernating" | .status is not "True"`) doesn't appear to be working, I claimed a cluster successfully and then checked the `status.conditions` array of the ClusterDeployment and saw that the cluster's `.status.conditions` array showed "Hibernating" as "True" (and a new discovery, another field called "Unreachable" also set to "true").  

### Debugging Info
Just claimed a cluster from our canary clusterpools:
```
gbuchana-mac:build-harness gurnben$ make clusterpool/list-clusterpools
NAME                READY   SIZE   BASEDOMAIN                 IMAGESET
canary-aws-4331     1       1      aws.red-chesterfield.com   openshift-v4331
canary-aws-4333     1       1      aws.red-chesterfield.com   openshift-v4333
canary-aws-4416     2       2      aws.red-chesterfield.com   openshift-v4416
canary-aws-4418     2       2      aws.red-chesterfield.com   openshift-v4418
canary-aws-455      2       2      aws.red-chesterfield.com   openshift-v455
canary-aws-457      2       2      aws.red-chesterfield.com   openshift-v457
canary-aws-460fc8   1       1      aws.red-chesterfield.com   openshift-v460fc8
make clusterpool/checkout CLUSTERPOOL_NAME=canary-aws-457 CLUSTERPOOL_CLUSTER_CLAIM=gurnben-pr-test
/Users/gurnben/Documents/Projects/build-harness/build-harness/vendor/oc apply -f /Users/gurnben/Documents/Projects/build-harness/build-harness/../build-harness-extensions/clusterclaim.yaml 
clusterclaim.hive.openshift.io/gurnben-pr-test created
ClusterClaimed
```

I then immediately grabbed the clusterclaim:
```
gbuchana-mac:build-harness gurnben$ oc get clusterclaim
NAME                    AGE
canary-aws-4333-nf0vf   18s
canary-aws-455-nf0vf    16s
david-claim2-aws-457    9h
gurnben-pr-test         31s
gbuchana-mac:build-harness gurnben$ oc get clusterclaim gurnben-pr-test -o yaml
...
spec:
  baseDomain: aws.red-chesterfield.com
  clusterMetadata:
    adminKubeconfigSecretRef:
      name: canary-aws-457-w789h-0-k7g2s-admin-kubeconfig
    adminPasswordSecretRef:
      name: canary-aws-457-w789h-0-k7g2s-admin-password
    clusterID: ee45e0f5-a8f0-4d93-a1cf-0da6475ec1ee
    infraID: canary-aws-457-w789h-mtk56
  clusterName: canary-aws-457-w789h
  clusterPoolRef:
    claimName: gurnben-pr-test
    namespace: clusterpool
    poolName: canary-aws-457
  controlPlaneConfig:
    servingCertificates: {}
  installed: true
  platform:
    aws:
      credentialsSecretRef:
        name: canary-aws-457-w789h-aws-creds
      region: us-east-1
  powerState: Running
  provisioning:
    imageSetRef:
      name: openshift-v457
    installConfigSecretRef:
      name: canary-aws-457-w789h-install-config
  pullSecretRef:
    name: canary-aws-457-w789h-pull-secret
...
```
(collapsed for conciseness)

Then I extracted the namespacce and took a look at the clusterdeployment:
```
gbuchana-mac:build-harness gurnben$ oc get clusterdeployment canary-aws-457-w789h -n canary-aws-457-w789h -o yaml
...
  conditions:
  - lastProbeTime: "2020-09-29T12:00:41Z"
    lastTransitionTime: "2020-09-29T04:29:50Z"
    message: Starting cluster machines
    reason: Resuming
    status: "True"
    type: Hibernating
  - lastProbeTime: "2020-09-29T04:29:49Z"
    lastTransitionTime: "2020-09-29T04:29:49Z"
    message: SyncSet apply is successful
    reason: SyncSetApplySuccess
    status: "False"
    type: SyncSetFailed
  - lastProbeTime: "2020-09-29T11:28:25Z"
    lastTransitionTime: "2020-09-29T04:29:52Z"
    message: 'Get https://api.canary-aws-457-w789h.aws.red-chesterfield.com:6443/api?timeout=32s:
      dial tcp 54.198.144.48:6443: i/o timeout'
    reason: ErrorConnectingToCluster
    status: "True"
    type: Unreachable
...
```
Uh oh...

Tried to log in with the credentials extracted by get-cluster-metadata:
```
gbuchana-mac:build-harness gurnben$ make clusterpool/get-cluster-metadata CLUSTERPOOL_NAME=canary-aws-457 CLUSTERPOOL_CLUSTER_CLAIM=gurnben-pr-test CLUSTERPOOL_METADATA_FILE=gurnben-pr-test.json
{
  "username": "kubeadmin",
  "password": "REDACTED",
  "basedomain": "canary-aws-457-w789h.aws.red-chesterfield.com",
  "api_url": "https://api.canary-aws-457-w789h.aws.red-chesterfield.com:6443",
  "console_url": "https://console-openshift-console.apps.canary-aws-457-w789h.aws.red-chesterfield.com"
}
gbuchana-mac:build-harness gurnben$ oc login https://api.canary-aws-457-w789h.aws.red-chesterfield.com:6443 -u kubeadmin -p REDACTED
The server uses a certificate signed by an unknown authority.
You can bypass the certificate check, but any data you send to the server could be intercepted by others.
Use insecure connections? (y/n): y

Unable to connect to the server: EOF
gbuchana-mac:build-harness gurnben$ oc login https://api.canary-aws-457-w789h.aws.red-chesterfield.com:6443 -u kubeadmin -p REDACTED --insecure-skip-tls-verify=true
Unable to connect to the server: EOF
```

Then took a peak at the clusterdeployment in the namespace extracted from the clusterdeployment:
```
oc get clusterdeployment canary-aws-457-w789h -n canary-aws-457-w789h -o yaml
...
conditions:
  - lastProbeTime: "2020-09-29T12:00:41Z"
    lastTransitionTime: "2020-09-29T04:29:50Z"
    message: Starting cluster machines
    reason: Resuming
    status: "True"
    type: Hibernating
  - lastProbeTime: "2020-09-29T04:29:49Z"
    lastTransitionTime: "2020-09-29T04:29:49Z"
    message: SyncSet apply is successful
    reason: SyncSetApplySuccess
    status: "False"
    type: SyncSetFailed
  - lastProbeTime: "2020-09-29T12:01:40Z"
    lastTransitionTime: "2020-09-29T04:29:52Z"
    message: 'Get https://api.canary-aws-457-w789h.aws.red-chesterfield.com:6443/api?timeout=32s:
      net/http: TLS handshake timeout'
    reason: ErrorConnectingToCluster
    status: "True"
    type: Unreachable
...
```

It appears the cluster checkout exited before the cluster was available, let's make sure it _does_ become available...

Approximately 5 minutes later, I have _some_ good news to report!
```
gbuchana-mac:build-harness gurnben$ oc get clusterdeployment canary-aws-457-w789h -n canary-aws-457-w789h -o json | jq '.status.conditions'   [
  {
    "lastProbeTime": "2020-09-29T12:05:16Z",
    "lastTransitionTime": "2020-09-29T12:05:16Z",
    "message": "All machines are started and nodes are ready",
    "reason": "Running",
    "status": "False",
    "type": "Hibernating"
  },
  {
    "lastProbeTime": "2020-09-29T04:29:49Z",
    "lastTransitionTime": "2020-09-29T04:29:49Z",
    "message": "SyncSet apply is successful",
    "reason": "SyncSetApplySuccess",
    "status": "False",
    "type": "SyncSetFailed"
  },
  {
    "lastProbeTime": "2020-09-29T12:05:25Z",
    "lastTransitionTime": "2020-09-29T12:05:25Z",
    "message": "cluster is reachable",
    "reason": "ClusterReachable",
    "status": "False",
    "type": "Unreachable"
  }
]
```

I'm not sure if the status checks are working given the above.  

We may also want to add to the original git issue criteria to validate that `the .status.conditions | select(.type == "Unreachable" | .status is not "True"`.

**Review by gurnben — APPROVED** [review]
Tested locally and in canary, looks great and works beautifully @schmidtd, thank you for all the hard work and iteration!
